src/main/java/io/cdap/plugin/snowflake/source/batch/SnowflakeMapToRecordTransformer.java

```diff
@@ -55,7 +56,8 @@ private StructuredRecord getStructuredRecord(Map<String, String> row, Schema sch
       .filter(entry -> schema.getField(entry.getKey()) != null) // filter absent fields in the schema
       .forEach(entry -> builder.set(
         entry.getKey(),
-        convertValue(entry.getKey(), entry.getValue(), schema.getField(entry.getKey()).getSchema())));
+        convertValue(entry.getKey(), entry.getValue(),
+          Objects.requireNonNull(schema.getField(entry.getKey())).getSchema())));
```

Comment:
Shouldn't the method have `throws NullPointerException` in it's signature ?

Reply:
updated